stdlib/public/core/ExistentialCollection.swift.gyb

```diff
@@ -223,8 +244,15 @@ public struct AnySequence<Element> : SequenceType {
   @available(*, unavailable, renamed="Element")
   public typealias T = Element
 
-  /// Wrap and forward operations to to `base`.
-  public init<S: SequenceType where S.Generator.Element == Element>(_ base: S) {
+  /// Wrap and forward operations to `base`.
+  public init<
+    S: SequenceType
+    where
+      S.Generator.Element == Element,
+      S.SubSequence : SequenceType,
+      S.SubSequence.Generator.Element == Element,
+      S.SubSequence.SubSequence == S.SubSequence
+  >(_ base: S) {
```

Comment:
This changes the signature.  We want to constrain `SequenceType.SubSequence` this way anyway, but don't have the necessary language feature.  Nevertheless, this requires a writing a proposal in swift-evolution.